test/NuGet.Core.Tests/NuGet.Commands.Test/WarningPropertiesCollectionTests.cs

```diff
@@ -720,5 +739,44 @@ public void WarningPropertiesCollection_MessageWithTargetGraphAndDependencyWithN
             Assert.True(warningPropertiesCollection.ApplyWarningProperties(suppressedMessage));
             Assert.Equal(0, suppressedMessage.TargetGraphs.Count);
         }
+
+        [Theory]
+        [MemberData(nameof(WarningsNotAsErrorsTestCases))]
+        public void ApplyWarningProperties_WithProjectWarningsNotAsErrors(bool allWarningsAsErrors, HashSet<NuGetLogCode> noWarnSet, HashSet<NuGetLogCode> warnAsErrorSet, HashSet<NuGetLogCode> warningsNotAsErrors, LogLevel expectedLogLevel, bool suppressed)
```

Comment:
Just an idea, but if there's no `LogLevel.None`, then maybe `LogLevel?`, and using null to mean suppressed, will allow this method to have one fewer parameters?

Also, since there's only one NU code used for all tests, I think it's unnecessarily generic to pass in the 3 `HashSet`s as parameters. I don't know if it'll help readability, but maybe if there was a flags enum with values `None = 0, NoWarn = 0x01, AsError = 0x02, NotAsError = 0x04`, then the test cases would read something like `WarnPropertiesEnum.NoWarn | WarnPropertiesEnum.AsError`, rather than two new new hashsets.

Reply:
> Just an idea, but if there's no LogLevel.None, then maybe LogLevel?, and using null to mean suppressed, will allow this method to have one fewer parameters?

That's not enough, those are 2 different things. 
The method returns a value + updates the error message. Yeah, it's weird, but we need both to assert the correct behavior.

Reply:
> Also, since there's only one NU code used for all tests, I think it's unnecessarily generic to pass in the 3 HashSets as parameters. I don't know if it'll help readability, but maybe if there was a flags enum with values None = 0, NoWarn = 0x01, AsError = 0x02, NotAsError = 0x04, then the test cases would read something like WarnPropertiesEnum.NoWarn | WarnPropertiesEnum.AsError, rather than two new new hashsets.

I feel like that's way too much optimization for a single test case imo. 

The haset vs not a hashset is a fair point, but i'm also thinking that it'd be a lot easier to add test with the current structure of the test than any other one.

The current structure is *very* explicit.